pkg/git/fetch.go

```diff
@@ -18,56 +28,85 @@ type Fetcher struct {
 func (f Fetcher) Fetch(dir, gitURL, gitRevision, metadataDir string) error {
 	f.Logger.Printf("Cloning %q @ %q...", gitURL, gitRevision)
 
-	repository, err := git2go.InitRepository(dir, false)
+	repository, err := gogit.PlainInit(dir, false)
 	if err != nil {
 		return errors.Wrap(err, "initializing repo")
 	}
-	defer repository.Free()
 
-	remote, err := repository.Remotes.CreateWithOptions(parseURL(gitURL), &git2go.RemoteCreateOptions{
-		Name:  "origin",
-		Flags: git2go.RemoteCreateSkipInsteadof,
+	remote, err := repository.CreateRemote(&config.RemoteConfig{
+		Name: "origin",
+		URLs: []string{gitURL},
 	})
 	if err != nil {
 		return errors.Wrap(err, "creating remote")
 	}
-	defer remote.Free()
 
-	err = remote.Fetch([]string{"refs/*:refs/*"}, &git2go.FetchOptions{
-		DownloadTags: git2go.DownloadTagsAll,
-		RemoteCallbacks: git2go.RemoteCallbacks{
-			CredentialsCallback:      keychainAsCredentialsCallback(f.Keychain),
-			CertificateCheckCallback: certificateCheckCallback(),
-		},
-		ProxyOptions: git2go.ProxyOptions{
-			Type: git2go.ProxyTypeAuto,
+	cred, err := f.Keychain.Resolve(gitURL, "", CredentialTypeUserpass)
+	if err != nil {
+		return errors.Wrap(err, "getting cred for url")
+	}
+
+	auth, err := cred.Cred()
+	if err != nil {
+		return errors.Wrap(err, "getting auth for url")
+	}
+
+	if httpsProxy, exists := os.LookupEnv("HTTPS_PROXY"); exists {
+		url, err := url2.Parse(httpsProxy)
+		if err != nil {
+			return errors.Wrap(err, "parsing HTTPS_PROXY url")
+		}
+		proxyClient := &http.Client{
+			Transport: &http.Transport{
+				Proxy: http.ProxyURL(url),
+			},
+			Timeout: 15 * time.Second,
+			CheckRedirect: func(req *http.Request, via []*http.Request) error {
```

Comment:
Need to review appropriate values for Timeout, CheckRedirect, and all other fields of http.Client